PWGHF/HFC/Tasks/taskCorrelationLcHadrons.cxx

```diff
@@ -0,0 +1,271 @@
+  void processMcRec(LcHadronPairFull const& pairEntries)
+  {
+    for (const auto& pairEntry : pairEntries) {
+      // define variables for widely used quantities
+      double deltaPhi = pairEntry.deltaPhi();
+      double deltaEta = pairEntry.deltaEta();
+      double ptLc = pairEntry.ptLc();
+      double ptHadron = pairEntry.ptHadron();
+      double massLc = pairEntry.mLc();
+      double efficiencyWeight = 1.;
+      double efficiencyHadron = 1.;
+      int effBinLc = o2::analysis::findBin(binsPtEfficiency, ptLc);
+      int ptBinLc = o2::analysis::findBin(binsPtCorrelations, ptLc);
+      int poolBin = pairEntry.poolBin();
+      if (ptBinLc < 0 || effBinLc < 0) {
+        continue;
+      }
+      if (ptHadron > 10.0) {
+        ptHadron = 10.5;
+      }
+      if (applyEfficiency) {
+        efficiencyWeight = 1. / (efficiencyLc->at(effBinLc) * efficiencyHadron);
+      }
+      // fill correlation plots for signal/bagkground correlations
+      if (pairEntry.signalStatus()) {
+        registry.fill(HIST("hCorrel2DVsPtSignalMcRec"), deltaPhi, deltaEta, ptLc, ptHadron, poolBin, efficiencyWeight);
+      } else {
+        registry.fill(HIST("hCorrel2DVsPtBkgMcRec"), deltaPhi, deltaEta, ptLc, ptHadron, poolBin, efficiencyWeight);
+      }
+      // reject entries outside Pt ranges of interest
+
+      // check if correlation entry belongs to signal region, sidebands or is outside both, and fill correlation plots
+      if (massLc > signalRegionInner->at(ptBinLc) && massLc < signalRegionOuter->at(ptBinLc)) {
+        // in signal region
+        registry.fill(HIST("hCorrel2DVsPtSignalRegionMcRec"), deltaPhi, deltaEta, ptLc, ptHadron, poolBin, efficiencyWeight);
+        registry.fill(HIST("hCorrel2DPtIntSignalRegionMcRec"), deltaPhi, deltaEta, efficiencyWeight);
+        registry.fill(HIST("hDeltaEtaPtIntSignalRegionMcRec"), deltaEta, efficiencyWeight);
+        registry.fill(HIST("hDeltaPhiPtIntSignalRegionMcRec"), deltaPhi, efficiencyWeight);
+      }
+
+      if (((massLc > sidebandLeftOuter->at(ptBinLc)) && (massLc < sidebandLeftInner->at(ptBinLc))) ||
+          ((massLc > sidebandRightInner->at(ptBinLc) && massLc < sidebandRightOuter->at(ptBinLc)))) {
+        // in sideband region
+        registry.fill(HIST("hCorrel2DVsPtSidebandsMcRec"), deltaPhi, deltaEta, ptLc, ptHadron, poolBin, efficiencyWeight);
+        registry.fill(HIST("hCorrel2DPtIntSidebandsMcRec"), deltaPhi, deltaEta, efficiencyWeight);
+        registry.fill(HIST("hDeltaEtaPtIntSidebandsMcRec"), deltaEta, efficiencyWeight);
+        registry.fill(HIST("hDeltaPhiPtIntSidebandsMcRec"), deltaPhi, efficiencyWeight);
+      }
+    } // end loop
+  }
+  PROCESS_SWITCH(HfTaskCorrelationLcHadrons, processMcRec, "Process Mc Reco mode", false);
+
+  /// Lc-Hadron correlation pair filling task, from pair tables - for Mc gen-level analysis (no filter/selection, only true signal)
+  void processMcGen(aod::LcHadronPair const& pairEntries)
```

Comment:
What's the reason to keep MC gen and MC reco separated in two process functions? Do you need only one of the two in some cases?